src/SA/ldapsearch/entry.c

```diff
@@ -119,14 +166,22 @@ PLDAPSearch ExecuteLDAPQuery(LDAP* pLdapConnection, PCHAR distinguishedName, cha
     }
 
         
+    LDAPControlA simpleControl;
+    PLDAPControlA controlArray[2];
+
+    simpleControl = *FormatSDFlags(OWNER_SECURITY_INFORMATION | GROUP_SECURITY_INFORMATION | DACL_SECURITY_INFORMATION);
```

Comment:
While this works at a functional level, the memory returned by FormatSDFlags is leaked at this point.

We should return into a PLDAPControlA type variable and then use it directly in controlArray[0] instead of trying to return into a stack variable and then referring to it by reference again.

We are also missing a NULL check as FormatSDFlags can return NULL in its error cases, but the current code does not account for that.